listenbrainz/webserver/static/js/src/types.d.ts

```diff
@@ -31,6 +33,47 @@ declare interface ModalProps {
   onClose(event: React.MouseEvent<HTMLButtonElement>): void;
 }
 
+declare interface RecentListensProps {
+  api_url: string;
+  artist_count?: number | null | undefined;
+  // TODO: Change any to valid type
+  follow_list?: any;
+  follow_list_id?: string;
+  follow_list_name?: string;
+  have_listen_count?: boolean;
+  latest_listen_ts?: number;
+  latest_spotify_uri?: string;
+  listen_count?: number;
+  listens?: Listen[];
+  mode: "listens" | "follow";
+  next_listen_ts?: number;
+  previous_listen_ts?: number;
+  profile_url?: string;
+  save_url?: string;
+  spotify: {
+    access_token?: string;
+    permission?: string;
+  };
+  user: User;
+  web_sockets_server_url: string;
+}
+
+declare interface RecentListensState {
```

Comment:
I'd like to keep component specific props and states in the component file itself.

* We shouldn't need to import these types much anyways other than tests. 
* It is gonna be easier to read the props etc if they're in the file vs having to open a file which will potentially get huge if we put ALL types in it